ferveo/src/api.rs

```diff
@@ -0,0 +1,326 @@
+    g_inv: &G1Prepared,
+) -> Vec<u8> {
+    tpke::api::decrypt_with_shared_secret(
+        &ciphertext.0,
+        aad,
+        &shared_secret.0,
+        &g_inv.0,
+    )
+    .unwrap()
+}
+pub struct G1Prepared(pub tpke::api::TpkeG1Prepared);
+
+pub struct SharedSecret(tpke::api::TpkeSharedSecret);
+
+pub struct Keypair(ferveo_common::Keypair<E>);
+
+impl Keypair {
+    pub fn random<R: RngCore>(rng: &mut R) -> Self {
+        Self(ferveo_common::Keypair::<E>::new(rng))
+    }
+
+    pub fn public_key(&self) -> PublicKey {
+        PublicKey(self.0.public())
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        Self(ferveo_common::Keypair::<E>::deserialize(bytes).unwrap())
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut buf = vec![];
+        self.0.serialize(&mut buf).unwrap();
+        buf
+    }
+}
+
+#[derive(Clone)]
+pub struct PublicKey(ferveo_common::PublicKey<E>);
+
+impl PublicKey {
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        Self(ferveo_common::PublicKey::<E>::deserialize(bytes).unwrap())
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut buf = vec![];
+        self.0.serialize(&mut buf).unwrap();
+        buf
+    }
+}
+
+#[derive(Clone)]
+pub struct ExternalValidator(ferveo_common::ExternalValidator<E>);
+
+impl ExternalValidator {
+    pub fn new(address: String, public_key: PublicKey) -> Self {
+        Self(ferveo_common::ExternalValidator {
+            address,
+            public_key: public_key.0,
+        })
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Transcript(crate::PubliclyVerifiableSS<E>);
+
+impl Transcript {
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        Self(crate::PubliclyVerifiableSS::<E>::deserialize(bytes).unwrap())
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut buf = vec![];
+        self.0.serialize(&mut buf).unwrap();
+        buf
+    }
+}
+
+#[derive(Clone)]
+pub struct DkgPublicKey(pub tpke::api::TpkeDkgPublicKey);
+
+#[derive(Clone)]
+pub struct Dkg(crate::PubliclyVerifiableDkg<E>);
+
+impl Dkg {
+    pub fn new(
+        tau: u64,
+        shares_num: u32,
+        security_threshold: u32,
+        validators: &[ExternalValidator],
+        me: &ExternalValidator,
+    ) -> Self {
+        let validators = &validators
+            .iter()
+            .map(|v| v.0.clone())
+            .collect::<Vec<ferveo_common::ExternalValidator<E>>>();
+        let me = &me.0;
+        let params = crate::Params {
+            tau,
+            security_threshold,
+            shares_num,
+        };
+        let session_keypair = ferveo_common::Keypair::<E> {
+            decryption_key: ark_ff::UniformRand::rand(&mut ark_std::test_rng()),
+        };
+        let dkg = crate::PubliclyVerifiableDkg::<E>::new(
+            validators,
+            params,
+            me,
+            session_keypair,
+        )
+        .unwrap();
+        Self(dkg)
+    }
+
+    pub fn final_key(&self) -> DkgPublicKey {
```

Comment:
i think i'd prefer a name that made it clear this was a public key

Reply:
Could be just `public_key`